PurchasesCoreSwift/LocalReceiptParsing/ReceiptParser.swift

```diff
@@ -8,17 +8,32 @@
 
 import Foundation
 
-class ReceiptParser {
-    private let objectIdentifierParser: ASN1ObjectIdentifierBuilder
+@objc(RCReceiptParser) public class ReceiptParser: NSObject {
+    private let objectIdentifierBuilder: ASN1ObjectIdentifierBuilder
     private let containerBuilder: ASN1ContainerBuilder
     private let receiptBuilder: AppleReceiptBuilder
 
-    init(objectIdentifierParser: ASN1ObjectIdentifierBuilder = ASN1ObjectIdentifierBuilder(),
-         containerBuilder: ASN1ContainerBuilder = ASN1ContainerBuilder(),
-         receiptBuilder: AppleReceiptBuilder = AppleReceiptBuilder()) {
-        self.objectIdentifierParser = objectIdentifierParser
+    @objc public convenience override init() {
+        self.init(objectIdentifierBuilder: ASN1ObjectIdentifierBuilder(),
+                  containerBuilder: ASN1ContainerBuilder(),
+                  receiptBuilder: AppleReceiptBuilder())
+    }
+    
+    init(objectIdentifierBuilder: ASN1ObjectIdentifierBuilder,
+         containerBuilder: ASN1ContainerBuilder,
+         receiptBuilder: AppleReceiptBuilder) {
+        self.objectIdentifierBuilder = objectIdentifierBuilder
         self.containerBuilder = containerBuilder
         self.receiptBuilder = receiptBuilder
+        super.init()
+    }
+    
+    @objc public func receiptHasTransactions(receiptData: Data) -> Bool {
+        if let receipt = try? parse(from: receiptData) {
+            return receipt.inAppPurchases.count > 0
+        }
+        // if the receipt can't be parsed, conservatively return true
+        return true
```

Comment:
I'm not a huge fan of returning a `Bool` here, since there are technically three possible cases: 
- true
- false
- unknown (receipt can't be parsed). 

However, this isn't trivial to do in swift-objc interoperability: 
- you can either use a `throws`, which forces you to replace the `Bool` with `NSNumber` or an `inout` param, and the caller has to pass in an error, and then unwrap the `NSNumber` or pass in the `inout` param
- encode the return into an object that basically wraps a `Bool`, then use that to encode the value. 

They're not _bad_, but they did feel too complicated for what we're doing here. 

More info: https://stackoverflow.com/a/36673027/5074358

Reply:
if you're curious about what this would look like, I have a stash that I can upload into a branch for perspective